src/rebootstrap-raft/main.go

```diff
@@ -92,21 +97,22 @@ func (c *rebootstrapCommand) Init(args []string) error {
 
```

Comment:
what about adding a 
```
func (c *rebootstrapCommand) getJujuPath(subPath) string {
  # When we are running inside a confined snap /var/lib gets remounted to /var/lib/hostfs/var/lib
  # But users shouldn't need to be aware of this fact
  jujuBaseDir := c.jujuDir
  if c.inSnap and strings.HasPrefix(jujuBaseDir, "/var/lib") and not strings.HasPrefix(jujuBaseDir, "/var/lib/snapd/hostfs") {
    jujuBaseDir = filepath.Join("/var/lib/snapd/hostfs", jujuBaseDir[1:])
  }
  return filepath.Join(jujuBaseDir, subPath)
}

Reply:
And then we can set c.inSnap as part of initialization somewhere, but in tests we can then override it to test both in-snap and not-in-snap functionality

Reply:
I would also be ok with just having the final juju directory be stored on the rebootstrapCommand and passing that around, but mostly I want to make sure that when a user supplies the directory, we remap that for the fact that our view of the filesystem is different than the users.
Which overriding the default /var/lib/juju doesn't do.